### ISSUE TITLE
Bug: specs don't run

### ISSUE THREAD

**Issue by asterite** [jgaskins/redis#1]
```crystal
🍏 crystal spec
Showing last frame. Use --error-trace for full trace.

There was a problem expanding macro 'method_missing'

Code in src/client.cr:35:5

 35 | macro method_missing(call)
      ^
Called macro defined in src/client.cr:35:5

 35 | macro method_missing(call)

Which expanded to:

 > 1 | @pool.checkout do |connection|
 > 2 |   connection.xadd(_arg0, _arg1, foo: foo)
                    ^---
Error: no argument named 'foo'

Matches are:
 - Redis::Commands#xadd(key : String, id : String, maxlen = nil, data : Hash(String, String) = Hash(String, String).new)
```

I wanted to checkout the code and try optimizing it even more. For example those `read_line.to_i` and `read_line.to_i64` could be optimized by using `IO#peek` and parsing numbers manually from the slice (if possible). But without specs running I can't make sure I won't break the code.

**Comment by jgaskins** [comment]
Nice catch! Fixed in 74386c2.

**Comment by jgaskins** [comment]
`xadd` does need more coverage, especially since it now has 3 different signatures, but that commit should get you to where you can benchmark the parser. 🙂
